base/set.jl

```diff
@@ -141,6 +141,27 @@ function unique(f::Callable, C)
     out
 end
 
+"""
+    alldistinct(itr)
+
+Test if all values from `itr` are distinct, when compared with `isequal`.
```

Comment:
hm, spoke too soon maybe - technically this isn't testing using `isequal` until #9381 gets resolved, right?